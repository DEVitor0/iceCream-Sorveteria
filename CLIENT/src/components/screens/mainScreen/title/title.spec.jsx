import { render, screen } from '@testing-library/react';
import Title from './index';

describe('Title Component', () => {
  test('renders the correct text and matches snapshot', () => {
    const { asFragment } = render(<Title />);

    const titleElement = screen.getByText(/Escolha sua sobremesa/i);
    expect(titleElement).toBeInTheDocument();

    const emphasisElement = screen.getByText(/favorita/i);
    expect(emphasisElement).toBeInTheDocument();

    expect(titleElement).toHaveClass('main-screen__container__texts__title');

    expect(asFragment()).toMatchSnapshot();
  });

  test('applies the correct class when id is passed', () => {
    const { container } = render(<Title id="test-id" />);
    const divElement = container.querySelector('#test-id');
    expect(divElement).toHaveClass('looking-main');
  });

  test('does not apply the class when no id is passed', () => {
    render(<Title />);
    const divElement = screen.getByText(/Escolha sua sobremesa/i).parentElement;
    expect(divElement).not.toHaveClass('looking-main');
  });
});
