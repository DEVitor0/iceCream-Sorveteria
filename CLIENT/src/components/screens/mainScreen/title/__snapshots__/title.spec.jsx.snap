// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component renders the correct text and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <h1
      class="main-screen__container__texts__title"
    >
      Escolha sua sobremesa
    </h1>
    <span
      class="main-screen__container__texts__emphasis"
    >
      favorita.
    </span>
  </div>
</DocumentFragment>
`;
